### ISSUE TITLE
BCEloss in color_syncnet_train.py

### ISSUE THREAD

**Issue by YunjinPark** [Rudrabha/Wav2Lip#227]
Thank you for the great work!

I'm wondering why you used **BCEloss** instead of **BCEWithLogitsLoss** in **_color_syncnet_train.py_**.

```
logloss = nn.BCELoss()
def cosine_loss(a, v, y):
    d = nn.functional.cosine_similarity(a, v)
    loss = logloss(d.unsqueeze(1), y)
    return loss
```

I think "a", "v" can have element what is negative value.
And d(shape: B * 1) can also have negative value, right?
But, we have to give a positive values to BCELoss. 
Why don't you use BCEWithLogitsLoss? or BCELoss with sigmoid ?

**Comment by prajwalkr** [comment]
> And d(shape: B * 1) can also have negative value, right?

No. a, v are non-negative because they are relu-activated.

**Comment by YunjinPark** [comment]
Thank you. I missed it. 

Why do you use relu for last activation? Does relu perform better than sigmoid?
